lib/plugins/physics.js

```diff
@@ -12,7 +12,9 @@ module.exports = inject
 const PI = Math.PI
 const PI_2 = Math.PI * 2
 const PHYSICS_INTERVAL_MS = 50
-const PHYSICS_TIMESTEP = PHYSICS_INTERVAL_MS / 1000
+const PHYSICS_TIMESTEP = PHYSICS_INTERVAL_MS / 1000 // 0.05
+const PHYSICS_SETINTERVAL_MS = 10 // this runs the doPhysics() loop faster, not tickPhysics()
```

Comment:
That really sounds like a horrible hack, you are running this loop 5x time faster than it needs to be. Is it still because of the inaccurate timers on windows ? If so it would be better to fix the problem at its source instead of trying to hack around it.

Reply:
It isn't, I decided to read through that article, and they do run the loop faster (way too fast because they used a while loop instead of a setInterval).

The faster loop is used to get a more accurate doPhysics(), otherwise why aren't we just running the contents of tickPhysics() in a setInterval loop?

Reply:
> The faster loop is used to get a more accurate doPhysics(), otherwise why aren't we just running the contents of tickPhysics() in a setInterval loop?

The accumulator is here to compensate late intervals, running it faster do no improve accuracy, it just takes more event loop time...

Checking once every tick is the optimal interval time, if you go slower you'll accumulate delay, if you go faster you waste cpu.

Reply:
How does the accumulator compensate late intervals if it is only run once every tick? It just keeps adding delay until deltaSeconds reaches 0.1 in which it fires twice and repeats the cycle. Unless that is the intended way?

Reply:
yes that is exactly the intended way

Reply:
Ah, because the double-fire behavior seems to happen so often that anticheats pick up on it, and I wanted to reduce the amount of double firing.